crates/stc_ts_type_checker/tests/conformance/types/objectTypeLiteral/constructSignatures/constructSignaturesWithOverloads2.stats.rust-debug

```diff
@@ -1,6 +1,6 @@
 Stats {
     required_error: 2,
     matched_error: 0,
-    extra_error: 1,
+    extra_error: 2,
```

Comment:
This happens in this case https://tsplay.dev/w22b4w

`self.find_type` only finds the first interface so its reported as an error on the last line.

Reply:
If there's already logic for interface declaration merging I can try to solve this on this PR otherwise I think it's too much :skull_and_crossbones:

Reply:
There's code for it.

https://github.com/dudykr/stc/blob/d60916bff1e5d71cbb81c0dc443b1ae77f0eb7df/crates/stc_ts_file_analyzer/src/analyzer/decl_merging.rs

Reply:
Alright, I ll give it a try. Im converting back to draft and I ll see what I can do :P